### DISCUSSION TITLE
Creating database dumps now that Supabase is using Postgres 14

### DISCUSSION THREAD

**Discussion by leo-petrucci** [supabase/supabase#4345] [Questions] [answered]
HI everyone,

A couple of days ago I was able to easily create and restore backups with `pg_dump` and `psql`, but as of today I'm getting errors of a version mismatch between my client and supabase (12.9 to 14.1).

I updated postgres to version 14.

```
psql --version
psql (PostgreSQL) 14.1 (Ubuntu 14.1-2.pgdg20.04+1)
```

But `pg_dump` did not upgrade with everything else:
```
pg_dump --version
pg_dump (PostgreSQL) 12.9 (Ubuntu 12.9-2.pgdg20.04+1)
```

Completely lost. I've tried uninstalling postgres entirely and reinstalling but now I just get:

```
pg_dump --version      
Error: PostgreSQL version 12 is not installed
```

Anyone having the same issue?

**Comment by kk21** [comment]
Have u tried the steps here to install specifically "postgresql-14"
https://www.postgresql.org/download/linux/ubuntu/

**Reply by leo-petrucci** [reply]
I ran:
```
sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'

wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -

sudo apt-get update

sudo apt-get -y install postgresql-14
```

psql 14 is installed, running `pg_dump` still returns:
```
pg_dump --version
Error: PostgreSQL version 12 is not installed
```

**Reply by kk21** [reply]
Run "lsb_release -cs" in terminal.
You may need to specify an official ubuntu name if you are using an unofficial ubuntu distribution.

**Reply by leo-petrucci** [reply]
I'm running this in WSL, but pretty sure it's just the stock ubuntu:
```
lsb_release -cs
focal
```

**Reply by kk21** [reply]
sudo apt purge postgresql*
sudo apt install postgresql-14

**Reply by leo-petrucci** [reply]
Ran:
```
sudo apt purge 'postgresql*'
```
Purged everything.

Ran: 
```
sudo apt install postgresql-14
```

And:
```
pg_dump --version
Error: PostgreSQL version 12 is not installed
```

Idk what to do at this point

**Reply by kk21** [reply]
Download the windows installer if u are on windows?
https://www.postgresql.org/download/windows/

**Reply by bnjmnt4n** [reply]
Try installing `postgresql-client-14`. It seems as though `postgresql-14` only includes the database server, and not some of the other client utilities.

**Comment by leo-petrucci** [comment, marked as answer]
Found a solution!

https://dba.stackexchange.com/a/303633/219160

> The output of pg_lsclusters indicates that Postgres 12 binaries were uninstalled without removing the Postgres 12 instance first. Notice how the Status column says binaries_missing. The removal scripts takes special care not to destroy your data when you're uninstalling, which is why an instance and its data can be a bit sticky.
>
>To remove the "zombie" instance, use
>
>```
>sudo pg_dropcluster 12 main
>```
